database/database.go

```diff
@@ -292,6 +297,60 @@ func (db *Database) GetSoftwareUpdateState() (state json.RawMessage, err error)
 	return state, nil
 }
 
+func (db *Database) GetDownloadInfo(filePath string) (downloadInfo downloader.DownloadInfo, err error) {
+	if err = db.getDataFromQuery(
+		"SELECT * FROM download WHERE path = ?",
+		[]any{filePath}, &downloadInfo.Path, &downloadInfo.TargetType,
+		&downloadInfo.InterruptReason, &downloadInfo.Released); err != nil {
+		if errors.Is(err, errNotExist) {
+			return downloadInfo, downloader.ErrNotExist
+		}
+	}
+
+	return downloadInfo, err
+}
+
+func (db *Database) GetDownloadInfos() ([]downloader.DownloadInfo, error) {
+	return db.getDownloadFromQuery("SELECT * FROM download")
+}
+
+func (db *Database) RemoveDownloadInfo(filePath string) (err error) {
+	if err = db.executeQuery("DELETE FROM download WHERE path = ?", filePath); errors.Is(err, errNotExist) {
+		return nil
+	}
+
+	return err
+}
+
+func (db *Database) SetDownloadInfo(downloadInfo downloader.DownloadInfo) (err error) {
+	var path string
+
+	if err = db.getDataFromQuery(
+		"SELECT * FROM download WHERE path = ?",
+		[]any{downloadInfo.Path}, &path); err != nil {
+		if !errors.Is(err, errNotExist) {
+			return err
+		}
+	}
+
+	if path != "" {
```

Comment:
Checking errNotExist is more clear check then `path != ""`.

Reply:
Done